x-pack/plugin/esql/qa/testFixtures/src/main/resources/stats.csv-spec

```diff
@@ -663,6 +663,65 @@ ca:l | cx:l | l:i
 1    | 1    | null
 ;
 
+///////////////////////////////////////////////////////////////
+// Test edge case interaction with push down optimization rules
+// https://github.com/elastic/elasticsearch/issues/108008
+///////////////////////////////////////////////////////////////
+
+countSameFieldWithEval
+required_capability: fixed_pushdown_past_project
+from employees | stats  b = count(gender), c = count(gender) by gender | eval b = gender | sort c asc
```

Comment:
I understand the goal is to make the query run to completion, but the first time I looked at this query, the alias b looks ambiguous, should it be gender or count(gender)? Seems like we return b as gender, it overwrites count(gender), but when pushdown happens, the order might be reversed. In SQL, if users code ambiguous column references, an error will return. Should we return an error here to indicate that b is ambiguous or make it return successfully here(if we have agreement in ES|QL that if the same alias is defined in multiple places, the last one will take effect)?

Reply:
I don't think we should throw errors. Masking happens all the time, even a simple `| eval a = 1 | ... | eval a = 2` could be considered masking.
The intention is exactly to make sure that the final result for `b` is the value of `gender`, even if EVAL gets pushed down before STATS (that is what is supposed to happen with current planning rules)

Reply:
Thanks @fang-xing-esql - ESQL in general allows shadowing attribute names that have been available previously. Take a look at the `shadowing...` csv tests to see this in action.  (The tests exist for most commands, `eval.csv-spec` may be the most important one, though.) Some PRs ago, I also updated our docs to describe behavior in case of conflicting names.

The main idea is that we want to be able to compose expressions in eval, like `EVAL x = to_upper(field), x = concat(x, some_other_field)`.